server/app/services/applicant/ApplicantServiceImpl.java

```diff
@@ -506,17 +513,32 @@ private void stageNormalUpdates(
       } else {
         switch (type) {
           case CURRENCY_CENTS:
-            applicantData.putCurrencyDollars(currentPath, update.value());
+            try {
+              applicantData.putCurrencyDollars(currentPath, update.value());
+            } catch (IllegalArgumentException e) {
+              applicantData.maybeDelete(currentPath);
```

Comment:
question (blocking): why is maybeDelete necessary here?

Reply:
If the applicant were editing a previously answered question, the ApplicantData would have the previous value. When invalid data was provided, we store that input in the failedUpdates map. I thought it might be confusing to have both the previous value and the invalid value present on ApplicantData.

I don't feel strongly about that though, so could remove the call to maybeDelete. Do either of you have a preference either way?

Reply:
It gets a little trickier to think about in other scenarios. E.g. if we introduce validations that rely on looking at multiple pieces of data, clearing the value results in different behavior than leaving the invalid value. This gets even trickier when we consider that the bad update may be attempting to change a value that is already set and previously valid -- deleting it creates yet a different validation state if we have those kinds of cross-cutting validations.

I'm not sure what the best approach is, I'm very slightly leaning towards leaving the invalid update in place since it seems simplest for now, but if you feel different I'm totally open to other thoughts.

Reply:
Those are all reasonable concerns. I'll leave the old value in place for now. It's also worth calling out that I added logic to ApplicantData.toJsonString to ensure that we don't allow persisting invalid updates as a bit of defensive coding.

Reply:
Nice one, thanks for that!